src/TestFramework/TestFramework/Attributes/TestMethod/TimeoutAttribute.cs

```diff
@@ -35,4 +37,21 @@ public TimeoutAttribute(TestTimeout timeout)
     /// Gets the timeout in milliseconds.
     /// </summary>
     public int Timeout { get; }
+
+    /// <summary>
+    /// Gets or sets a value indicating whether the method should be cooperatively cancelled on timeout.
+    /// When set to <see langword="true"/>, the method should be designed to cooperatively cancel itself when the timeout is reached.
+    /// Otherwise, when set to <see langword="false"/>, the task method will be unobserved.
+    /// </summary>
```

Comment:
```suggestion
    /// Gets or sets a value indicating whether the test method should be cooperatively cancelled on timeout. 
    /// When set to <see langword="true"/>, the cancellation token is cancelled on timeout, and the method completion is awaited. The test method and all the code it calls, must be designed in a way that it observes the cancellation and cancels cooperatively. If the test method does not complete, the timeout does not force it to complete.
    /// When set to <see langword="false"/>, the cancellation token is cancelled on timeout, timeout result is reported and the method task will continue running on background. This may lead to conflicts in file access on test cleanup, unobserved exceptions, and memory leaks.
    /// </summary>
```